include/circt/Dialect/HW/HWStructure.td

```diff
@@ -66,6 +67,24 @@ class HWModuleOpBase<string mnemonic, list<Trait> traits = []> :
           return sym;
       return ::circt::hw::InnerSymAttr();
     }
+
+    void $cppClass::setPortSymbolAttr(size_t portIndex, circt::hw::InnerSymAttr symbol) {
+      for (NamedAttribute argAttr :
+           mlir::function_interface_impl::getArgAttrs(*this, portIndex))
+        if (auto sym = argAttr.getValue().dyn_cast<circt::hw::InnerSymAttr>()) {
+          if (symbol && !symbol.empty())
+            mlir::function_interface_impl::setArgAttr(*this, portIndex, argAttr.getName(), symbol);
+          else
+            mlir::function_interface_impl::removeArgAttr(*this, portIndex, argAttr.getName());
+          return;
+        }
+      // We cannot add symbols to arguments without assuming the attribute name,
```

Comment:
The get version (directly above) of this walks all named attributes for any of the InnerSymAttr type, which doesn't fit well with being able to /add/ new ones.  So here this does the same strategy, so that if we can /get/ the symbol for a port we can update it to drop a particular field's symbol, without adding a requirement not already present.